### PR TITLE
ranger: refactor ports usage

### LINKED ISSUE #513: ranger: refactor ports usage

### PR DESCRIPTION
**Pull request by mehdibn**
#### Which issue(s) this PR fixes:

Fixes #513 

This PR resolves mainly #513 that requires the following features:

- a new utils role for xml merge : `xml/merge`

#### Additional comments:

To make clear that you license your contribution under
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0) and that you give permission to [TOSIT](https://www.tosit.io/),
you have to acknowledge this by using the following check-box.

 - [X] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [X] I hereby agree to grant [TOSIT](https://www.tosit.io/) a copyright license to use my contributions.

### CODE REVIEW

File: roles/ranger/admin/tasks/config.yml [hunk 1]
```diff
@@ -31,3 +31,12 @@
...
     dest: '{{ ranger_install_dir }}/ews/ranger-admin-services.sh'
+
+- import_role:
+    name: tosit.tdp.utils.xml
+    tasks_from: merge
+  vars:
+    source_file: '{{ ranger_install_dir }}/conf/ranger-admin-site.xml'
+    merge_var: '{{ ranger_admin_site  }}'
+    dest_file: '{{ ranger_install_dir }}/conf/ranger-admin-site.xml'
+    with_xml_description: true
```
Comment (Nuttymoon) [inline roles/ranger/admin/tasks/config.yml:42]: The Ansible `xml` module is enough to replace a few properties in the file. Here I just replace the `value` tags:
```suggestion
- name: Overwrite configurations not set by setup.sh
  xml:
    path: "{{ ranger_install_dir }}/conf/ranger-admin-site.xml"
    xpath: "/configuration/property[name='{{ item.key }}']/value"
    value: "{{ item.value }}"
    pretty_print: yes
  loop: "{{ ranger_admin_site | dict2items }}"
```

Comment (mehdibn) [reply roles/ranger/admin/tasks/config.yml:42]: @Nuttymoon thank you very much for your help :)
I tried this module, and the problem is that the task fails when the property doesn't exist.

Comment (Nuttymoon) [reply roles/ranger/admin/tasks/config.yml:42]: Indeed (although I think that they exist by default in `ranger-admin-site.xml`). But you can check if properties exist and create them if not:
```yaml
- name: Check if properties to overwrite exist
  xml:
    path: "{{ ranger_install_dir }}/conf/ranger-admin-site.xml"
    xpath: "/configuration/property[name='{{ item }}']"
  loop: "{{ ranger_admin_site.keys() | list }}"
  failed_when: no
  register: setup_sh_overwrite_props

- name: Create missing properties to overwrite
  xml:
    path: "{{ ranger_install_dir }}/conf/ranger-admin-site.xml"
    xpath: /configuration
    add_children:
      - property:
          _:
            - name: "{{ item.item }}"
            - value: "{{ ([item.item] | map('extract', ranger_admin_site) | list)[0] }}"
    pretty_print: yes
  loop: "{{ setup_sh_overwrite_props.results }}"
  when: '"Can''t process Xpath" in (item.msg | default(""))'
  register: setup_sh_overwrite_props
```

Comment (Nuttymoon) [reply roles/ranger/admin/tasks/config.yml:42]: It's a bit complicated but still better than introducing a new module IMO. Moreover this is temporary until we rewrite the `setup.sh`.

Comment (mehdibn) [reply roles/ranger/admin/tasks/config.yml:42]: thanks @Nuttymoon, indeed, i did something like this :) that's perfect, thank you again !

Comment (Nuttymoon) [reply roles/ranger/admin/tasks/config.yml:42]: I simplified the snippet up there!

Comment (mehdibn) [reply roles/ranger/admin/tasks/config.yml]: perfect :) Many thanks @Nuttymoon !!

File: roles/utils/xml/tasks/merge.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    path: "{{ configuration_file }}"
+    xpath: "/configuration"
+    add_children:
+      - property:
+          _:
+            - name: "{{ item.item.key }}"
+            - value: "{{ item.item.value }}"
+    pretty_print: yes
+  loop: "{{ overwrite_results.results }}"
+  when: item.failed
```
Comment (Nuttymoon) [inline roles/utils/xml/tasks/merge.yml:26]:
```suggestion
- name: Add new properties
  xml:
    path: "{{ configuration_file }}"
    xpath: "/configuration"
    add_children:
      - property:
          _:
            - name: "{{ item.item.key }}"
            - value: "{{ item.item.value }}"
    pretty_print: yes
  loop: "{{ overwrite_results.results }}"
  loop_control:
    label: "{{ item.item }}"
  when: item.failed
```

**Review by Nuttymoon — APPROVED** [review]

**Review by gboutry — APPROVED** [review]

**Review by leopaul36 — APPROVED** [review]

**Pull request merged**